test/langtools/jdk/javadoc/doclet/testDirectedInheritance/TestDirectedInheritance.java

```diff
@@ -0,0 +1,767 @@
+                "x");
+        checkExit(Exit.OK);
+        new OutputChecker("x/E1.html").check("""
+                <div class="block">E1: main description
+                 I2: main description
+                 I1: main description</div>""", """
+                <dt>Throws:</dt>
+                <dd><code>F</code> - E1: description of an exception</dd>
+                <dd><code>F</code> - I2: first description of an exception</dd>
+                <dd><code>F</code> - I2: second description of an exception</dd>
+                <dd><code>F</code> - I1: first description of an exception</dd>
+                <dd><code>F</code> - I1: second description of an exception</dd>
+                </dl>""");
+        new OutputChecker(Output.OUT).setExpectFound(false)
+                .check("warning: not a direct supertype"); // no unexpected warnings
+    }
+
+    /*
+     * C1.m inherits documentation from B1.m, which does not override, but inherits A.m.
+     * C2.m inherits documentation from B2.m, which overrides A.m and implicitly inherits
+     * its documentation.
+     */
+    @Test
+    public void testRecursiveInheritance1(Path base) throws Exception {
+        Path src = base.resolve("src");
+        tb.writeJavaFiles(src, """
+                package x;
+                public class A {
+                    /** A.m() */
+                    public void m() { }
+                }
+                """, """
+                package x;
+                public class B1 extends A { }
+                """, """
+                package x;
+                public class C1 extends B1 {
+                    /** {@inheritDoc B1} */
+                    @Override public void m() { }
+                }
+                """, """
+                package x;
+                public class B2 extends A {
+                    @Override public void m() { }
+                }
+                """, """
+                package x;
+                public class C2 extends B2 {
+                    /** {@inheritDoc B2} */
+                    @Override public void m() { }
+                }
+                """);
+        javadoc("-d", base.resolve("out").toString(),
+                "--source-path", src.toString(),
+                "x");
+        checkExit(Exit.OK);
+        var m = """
+                <section class="detail" id="m()">
+                <h3>m</h3>
+                <div class="member-signature"><span class="modifiers">\
+                public</span>&nbsp;<span class="return-type">void</span>\
+                &nbsp;<span class="element-name">m</span>()</div>
+                <div class="block">A.m()</div>""";
+        new OutputChecker("x/C1.html").check(m);
+        new OutputChecker("x/C2.html").check(m);
+        new OutputChecker(Output.OUT).setExpectFound(false)
+                .check("warning: not a direct supertype"); // no unexpected warnings
+    }
+
+    /*
+     * C1.m inherits documentation from B1.m explicitly and undirect.
```

Comment:
possible typo: do you mean "indirect"

Reply:
I'm neither a grammarian nor a native English speaker, but my hunch is that _indirect_ would be wrong here. But again, I'm open to corrections.

I believe "directed inhertiance" came from an analogy with the graph theory, where a graph can be directed and undirected. The former has pointy arrows, the latter does not.

By that analogy, documentation inheritance can be directed `{@inheritDoc <FQN>}` (I choose), or undirected `{@inheritDoc}` (JavaDoc chooses / I'm Feeling Lucky). In addition to that, documentation inheritance can be implicit (i.e. by omission), which works the same way as undirected but without extra keystrokes.

While those concepts are lacking better names, can I suggest we use the following almost-made-up adverbs: _directedly_ and _undirectedly_? As a side note, we badly need a JavaDoc vocabulary.

Reply:
Have a look at this commit: 3da4f473a7c